telink: b92: fix the define .

haiwentelink opened the pull request:
- fix the IRQ_GPIO define .
- update west.yml .

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/34d2aacefc0c4bc9dc994b7f043f11f6cae30bf6 | https://github.com/telink-semi/hal_telink/commit/8e22fad3c938f4ea2266293897d8b8ea47e566a0 (develop) | [telink-semi/hal_telink@34d2aace..8e22fad3](https://github.com/telink-semi/hal_telink/compare/34d2aacefc0c4bc9dc994b7f043f11f6cae30bf6..8e22fad3c938f4ea2266293897d8b8ea47e566a0) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

haiwentelink committed:
telink: b92: adopt ble v4.0.3.0 and 4m flash support .

- change the b92 lib part .
- Start with 2-swire SPI APIs;
- Check and enable 4-swire SPI APIs in soc_b9x_check_flash during booting.
- add the k_timer to avoid continous lock and unlock.
- save time for the flash_b9x_erase part .
- add the flash all area protect for the read, write, erase .
- add the k_timer to avoid continous lock and unlock.
- save time for the flash_b9x_erase part .

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>

andriy-bilynskyy commented:
Telink HAL revision is 8e22fad3c938f4ea2266293897d8b8ea47e566a0
Plese update submodule

haiwentelink commented:
already changed the commit id , and ci also can pass .
